python/cudf/cudf/utils/ioutils.py

```diff
@@ -955,6 +958,47 @@ def get_filepath_or_buffer(
     return path_or_data, compression
 
 
+def get_writer_filepath_or_buffer(path_or_data, mode, **kwargs):
+    """
+    Return either a filepath string to data,
+    or a open file object to the output filesystem
+
+    Parameters
+    ----------
+    path_or_data : str, file-like object, bytes, ByteIO
+        Path to data or the data itself.
+    mode : str
+        Mode in which file is opened
+
+    Returns
+    -------
+    filepath_or_buffer : str,
+        Filepath string or buffer of data
+    should_close : bool
+        Whether filepath_or_buffer should be closed after IO
```

Comment:
This feels very unsafe / leaky in that it's dependent on us to make sure we call `is_close()` as appropriate instead of relying on reference counting semantics.

What if we returned a class from here instead of a string or buffer that automatically closes when it goes out of scope?

Reply:
Alternatively if that class isn't feasible, we should use a context manager to handle closing automatically instead.

Reply:
Agreed. I'll look into a class or context manager that doesn't rely on the individual writers to _remember_ to close the buffer

Reply:
`fs.open` returns an `fsspec.openFile` object which returns the underlying file object when calling `.open` on this class object, or using a context like `with openFileobj as f: ...`

Looks like we can default to returning the `fsspec.openFile` object here and have all writers do something like
```
if path_or_buf is openFile object
with path_or_buf as f:
write...
```

Reply:
👍